tests/element_test_vectors.rs

```diff
@@ -34,60 +37,129 @@ fn contains_path(paths: &[&str], file_name: &str) -> bool {
 // Statically defined arrays containing test file paths to skip in various contexts.
 type SkipList = &'static [&'static str];
 
+// TODO: This trait exists solely for the purposes of this integration test. It allows us to paper
+// over the differences in the ion-c and native Rust ElementWriter implementations.
+// Once all of the tests are passing, we should work to unify their APIs.
+trait RoundTrip {
+    /// Encodes `elements` to a buffer in the specified Ion `format` and then reads them back into
+    /// a `Vec<OwnedElement>` using the provided reader.
+    fn roundtrip<R: ElementReader>(
+        elements: &Vec<OwnedElement>,
+        format: Format,
+        reader: R,
+    ) -> IonResult<Vec<OwnedElement>>;
+}
+
+/// These unit structs implement the [RoundTrip] trait and can be passed throughout the
+/// integration tests to dictate which readers/writers should be used in each test.
+struct IonCElementWriterApi;
+struct NativeElementWriterApi;
+
+impl RoundTrip for IonCElementWriterApi {
+    fn roundtrip<R: ElementReader>(
+        elements: &Vec<OwnedElement>,
+        format: Format,
+        reader: R,
+    ) -> IonResult<Vec<OwnedElement>> {
+        let mut buffer = vec![0; WRITE_BUF_LENGTH];
+        let mut ion_c_element_writer = format.element_writer_for_slice(buffer.as_mut_slice())?;
+        ion_c_element_writer.write_all(elements)?;
+        let bytes = ion_c_element_writer.finish()?;
+        reader.read_all(bytes)
+    }
+}
+
+// TODO: This implementation is not yet being used; the native reader still paired with the ion-c
+//       writer in the interest of minimizing the number of places that test failures need to be
+//       researched. Once the native reader is passing enough ion-tests, we'll enable the
+//       native writer so we can fix any failures it produces.
+impl RoundTrip for NativeElementWriterApi {
+    fn roundtrip<R: ElementReader>(
+        elements: &Vec<OwnedElement>,
+        format: Format,
+        reader: R,
+    ) -> IonResult<Vec<OwnedElement>> {
+        // Unlike the C writer, the Rust writer can write into a growing Vec.
+        // No need for an aggressive preallocation.
+        let mut buffer = Vec::with_capacity(2048);
+        match format {
+            // TODO: Handle `TextKind`: Pretty
```

Comment:
🗺️ `ion-rust` does not yet have a pretty printer. It should be relatively easy to implement, however, now that we have a `Writer` trait.